### PR TITLE
[PPP-3405] transformations / ktr exported from BA server contain all BA datasources

### PR DESCRIPTION
**Pull request by SergeyTravin**

### CODE REVIEW

**Comment by SergeyTravin** [comment]
Should be merged before https://github.com/pentaho/pdi-ee-plugin/pull/264

**Comment by SergeyTravin** [comment]
@mattcasters,  @brosander, @mattyb149 Could you review this, please?

File: engine/src/org/pentaho/di/trans/TransMeta.java [hunk 1]
```diff
@@ -2928,6 +2937,7 @@ public void loadXML( Node transnode, String fname, IMetaStore metaStore, Reposit
 
         // Handle connections
         int n = XMLHandler.countNodes( transnode, DatabaseMeta.XML_TAG );
+        List<String> privateTransormationDatabases = new ArrayList<String>();
```
Comment (rpbouman) [inline engine/src/org/pentaho/di/trans/TransMeta.java]: Typo: privateTransormation -> privateTransformation

File: engine/src/org/pentaho/di/trans/TransMeta.java [hunk 2]
```diff
@@ -2939,6 +2949,9 @@ public void loadXML( Node transnode, String fname, IMetaStore metaStore, Reposit
 
           DatabaseMeta dbcon = new DatabaseMeta( nodecon );
           dbcon.shareVariablesWith( this );
+          if ( !dbcon.isShared() ) {
+            privateTransormationDatabases.add( dbcon.getName() );
```
Comment (rpbouman) [inline engine/src/org/pentaho/di/trans/TransMeta.java]: Typo: privateTransormation -> privateTransformation

File: engine/src/org/pentaho/di/trans/TransMeta.java [hunk 3]
```diff
@@ -2956,6 +2969,7 @@ public void loadXML( Node transnode, String fname, IMetaStore metaStore, Reposit
             }
           }
         }
+        setPrivateTransformationDatabases( privateTransormationDatabases );
```
Comment (rpbouman) [inline engine/src/org/pentaho/di/trans/TransMeta.java]: Typo: privateTransormation -> privateTransformation

**Pull request merged**